service/src/main/java/uk/nhs/adaptors/gp2gp/ehr/mapper/AllergyStructureMapper.java

```diff
@@ -187,4 +189,11 @@ private String buildNotePertinentInformation(AllergyIntolerance allergyIntoleran
         }
         return notes;
     }
+
+    private String buildCode(AllergyIntolerance allergyIntolerance) {
+        if (allergyIntolerance.hasCode()) {
+            return codeableConceptCdMapper.mapCodeableConceptToCd(allergyIntolerance.getCode());
+        }
+        return StringUtils.EMPTY;
```

Comment:
Where properties are required (Card 1..1) the adaptor should throw an exception if not present.